src/main/java/org/folio/circulation/domain/CreateRequestService.java

```diff
@@ -130,4 +136,26 @@ private static RequestAndRelatedRecords setRequestQueuePosition(
       return succeeded(requestAndRelatedRecords);
     }
   }
+
+  public static Result<RequestAndRelatedRecords> refuseWhenUserHasAlreadyRequestedItem(RequestAndRelatedRecords request) {
+
+    Optional<Request> requestOptional = request.getRequestQueue().getRequests()
+      .stream()
+      .filter(it -> isTheSameRequester(request, it) && it.isOpen())
+      .findFirst();
+
+    if (requestOptional.isPresent()) {
+      Map<String, String> parameters = new HashMap<>();
+      parameters.put("requesterId", requestOptional.get().getUserId());
```

Comment:
These parameter values should be from the incoming request. Are we using the values from the matching request because that is more convenient?

```suggestion
      parameters.put("requesterId", request.getRequest().getUserId());
```

Reply:
Do you mind if I will make like this
`      parameters.put("requestId", requestOptional.get().getId());
      parameters.put("requesterId", request.getRequest().getUserId());
      parameters.put("itemId", request.getRequest().getItemId());`

Reply:
Sure, we don't really have guidelines for parameters. Can the `requestId` go last in the parameter list.

I've based most of my guidance on opinions shared about JSON API and how they should refer to input parameters, `requestId` does not fulfil that, but it might be useful diagnostic information.

Reply:
Done